docs/data-sources/protectionpolicy.md

```diff
@@ -0,0 +1,117 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
```

Comment:
why these docs changes are there in this PR ??

Reply:
these are auto generated docs and were pending to be merged for newly implemented resources/data-sources so, added them all in the PR.

Reply:
In my opinion not a very good approach to add all in one PR . we should generate docs for the implemented resource/data-source in their respective PR instead of adding them all in one. For now It's fine, But I'll not recommend this for future PR's. Please let me know If you think otherwise.

Reply:
agreed